evalml/pipelines/components/transformers/imputers/imputer.py

```diff
@@ -63,8 +64,8 @@ def fit(self, X, y=None):
         Returns:
             self
         """
-        X = _convert_to_woodwork_structure(X)
-        cat_cols = list(X.select('category').columns)
+        X = infer_feature_types(X)
+        cat_cols = list(X.select(['category', 'boolean']).columns)
```

Comment:
I can double check we're testing this but IIRC, it's so that we treat boolean columns like categorical cols. We don't want to pass it to our numeric_imputer which might try to use "mean" as the impute strategy and end up with nonsense for the imputed values.